SampleApplications/SDK/Opc.Ua.Client/Session.cs

```diff
@@ -2011,25 +2024,27 @@ public ReferenceDescriptionCollection FetchReferences(NodeId nodeId)
                 requireEncryption = identityPolicy.SecurityPolicyUri != SecurityPolicies.None;
             }
 
-            // validate the server certificate.
+            // validate the server certificate /certificate chain.
             X509Certificate2 serverCertificate = null;
             byte[] certificateData = m_endpoint.Description.ServerCertificate;
+
             if (certificateData != null && certificateData.Length > 0 && requireEncryption)
             {
-                serverCertificate = Utils.ParseCertificateBlob(certificateData);
-                m_configuration.CertificateValidator.Validate(serverCertificate);
+                X509Certificate2Collection serverCertificateChain = Utils.ParseCertificateChainBlob(certificateData);
+
+                if (serverCertificateChain.Count > 0)
+                {
+                    serverCertificate = serverCertificateChain[0];
+                }
+
+                m_configuration.CertificateValidator.Validate(serverCertificateChain);
```

Comment:
>m_configuration.CertificateValidator.Validate(serverCertificateChain); [](start = 16, length = 70)

only validate if count > 0?

Reply:
CertificateValidator.Validate should throw an exception if count = 0.
And that is ok because it validates that "certificateData" contains at least one certificate.